### PR TITLE
allow number formatting

### PR DESCRIPTION
**Pull request by Suyash-d**
**Is your feature request related to a problem? Please describe.**
I want a new instruction that allows me to reformat numbers based on a give target format.
* name: number.format
* parameters:
  * number: text, the number that needs to be reformatted
  * format: text, the format in Java String#format syntax (https://dzone.com/articles/java-string-format-examples section Formatting an Integer)
* return: the textual representation of the number reformatted with the given format

### CODE REVIEW

File: core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormat.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NumberFormat implements DataInstruction {
+
+    private final static String NUMBER = "number";
+    private final static String FORMAT = "format";
+
+    private final static Pattern PATTERN = Pattern.compile("\\s*\"(?<" + NUMBER + ">.+)\"\\s*,\\s*\"(?<" + FORMAT + ">.+)\"");
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormat.java:13]: can you make the double quotes surrounding the NUMBER optional

File: core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormatTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        NumberFormat numberFormat = new NumberFormat();
+        assertThrows(NumberFormatException.class, () -> numberFormat.generateOutput("\"25a\", \"|%d|\""));
+    }
+
+    @Test
+    void numberFormatThrowIllegalArgument() {
+        NumberFormat numberFormat = new NumberFormat();
+        assertThrows(IllegalArgumentException.class, () -> numberFormat.generateOutput("25a, |%d|"));
+    }
+
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormatTest.java:51]: can you test without the |

File: core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormat.java (hunk 1 above)
Comment (Suyash-d) [reply core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormat.java]: Done

File: core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormatTest.java (hunk 2 above)
Comment (Suyash-d) [reply core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormatTest.java:51]: Added

Comment (BerrevoetsRobbe) [reply core/java/iesi-core/src/test/java/io/metadew/iesi/script/execution/instruction/data/number/NumberFormatTest.java:51]: Please implement for all of the tests. Also write the tests without the double quotes

**Review by BerrevoetsRobbe — COMMENTED** [review]
approved

**Pull request merged**